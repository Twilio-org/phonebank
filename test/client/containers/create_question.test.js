import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import CreateQuestionContainer from '../../../public/src/containers/create_question';
import { mockStore } from '../client_test_helpers';

describe('<CreateQuestionContainer />', () => {
  const store = mockStore();
  describe('rendering', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <Provider store={store}>
          <MemoryRouter>
            <CreateQuestionContainer />
          </MemoryRouter>
        </Provider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
