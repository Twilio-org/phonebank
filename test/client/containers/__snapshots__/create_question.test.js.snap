// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateQuestionContainer /> rendering renders correctly 1`] = `
<div
  className="admin-form row"
>
  <div
    className="col-md-6 col-md-offset-3 col-xs-12"
  >
    <div
      className="page-header"
    >
      <h1>
        Create a Question
      </h1>
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-8"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                  htmlFor={undefined}
                >
                  Title
                </label>
                <span
                  className="help-block"
                >
                  For internal use.
                </span>
                <input
                  className="form-control"
                  id={undefined}
                  name="title"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Title"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="col-xs-4"
            >
              
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-xs-8"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                  htmlFor={undefined}
                >
                  Description
                </label>
                <span
                  className="help-block"
                >
                  What the volunteers will see.
                </span>
                <textarea
                  className="form-control"
                  id={undefined}
                  name="description"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  placeholder="Description"
                  type=""
                  value=""
                />
              </div>
            </div>
            <div
              className="col-xs-4"
            >
              
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col-xs-8"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                  htmlFor={undefined}
                >
                  Type
                </label>
                <select
                  className="form-control"
                  id={undefined}
                  name="type"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onDragStart={[Function]}
                  onDrop={[Function]}
                  onFocus={[Function]}
                  type={undefined}
                  value=""
                >
                  <option
                    hidden={true}
                    value="select"
                  >
                    Select
                  </option>
                  <option
                    id="paragraph"
                    value="paragraph"
                  >
                    Paragraph
                  </option>
                  <option
                    id="singleselect"
                    value="singleselect"
                  >
                    Multiple Choice
                  </option>
                  <option
                    id="multiselect"
                    value="multiselect"
                  >
                    Checkboxes
                  </option>
                </select>
              </div>
            </div>
            <div
              className="col-xs-4"
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        />
      </div>
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="submit"
        >
          Create Question
        </button>
        <button
          className="btn btn-default"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </form>
  </div>
</div>
`;
